packages/nhost-js/src/core/nhost-client.ts

```diff
@@ -40,13 +53,27 @@ export class NhostClient {
     devTools,
     start = true
   }: NhostClientConstructorParams) {
-    if (!backendUrl) throw new Error('Please specify a `backendUrl`. Docs: [todo]!')
+    if (!backendUrl && !subdomain)
+      throw new Error('Please specify either `backendUrl` or `subdomain`. Docs: [todo]!')
 
-    const subdomain = new URL(backendUrl).host.split('.')[0]
+    if (subdomain && subdomain !== 'localhost' && !region)
+      throw new Error('When using a subdomain, please specify a region.')
+
+    const urlFromEnv = (
+      backendUrl?: string,
+      subdomain?: string,
+      region?: string,
+      service?: string
+    ) => {
+      if (backendUrl) return `${backendUrl}/v1/${service}`
+
+      if (subdomain === 'localhost') return `http://${subdomain}`
```

Comment:
I'm not sure I understood. Assuming the CLI works in the same way as of today, and the user wants to use the CDN in cloud but to develop locally, his config would be something like
```js
{
  subdomain: 'localhost'
  region: 'eu-central-1' 
}
```

but then, local hasura-auth would be `http://localhost` while it should be `http://localhost:1337/v1/auth`